packages/portfolio/src/translators/reactjs/usePortfolioTokenTraits.tsx

```diff
@@ -0,0 +1,36 @@
+import {isRight} from '@yoroi/common'
+import {Chain, Portfolio} from '@yoroi/types'
+import {UseQueryOptions, useQuery} from 'react-query'
+
+export function usePorfolioTokenTraits(
+  {
+    id,
+    getTokenTraits,
+    network,
+  }: {
+    id: Portfolio.Token.Id
+    getTokenTraits: Portfolio.Api.Api['tokenTraits']
+    network: Chain.SupportedNetworks
+  },
+  options?: UseQueryOptions<
+    Portfolio.Token.Traits,
+    Error,
+    Portfolio.Token.Traits,
+    [Chain.SupportedNetworks, 'usePorfolioTokenTraits', Portfolio.Token.Id]
+  >,
+) {
+  const query = useQuery({
+    queryKey: [network, 'usePorfolioTokenTraits', id],
+    ...options,
+    queryFn: async () => {
+      const response = await getTokenTraits(id)
+      if (isRight(response)) return response.value.data
+      throw new Error('usePorfolioTokenTraits')
+    },
+  })
+
+  return {
+    ...query,
+    tokenTraits: query.data,
```

Comment:
If i am not wrong traits its a property of data right? `query.data.traits`

would it be a good idea to get the traits so you dont need to use `tokenTraits.traits`?

or does `query.data` contains more useful properties?

Reply:
ok i see it includes total items property

Reply:
```
 return {
    ...query,
    tokenTraits: query.data.traits,
    totalItems: query.data.totalItems,
 }
```

Maybe like this?

To avoid `tokenTraits.traits`

Reply:
yap, some info are not important cuz it does have the `info` before hitting it.